process/block/sovereignChainBlock.go

```diff
@@ -850,17 +857,79 @@ func (scbp *sovereignChainBlockProcessor) processSovereignBlockTransactions(
 		return nil, err
 	}
 
-	// TODO: Marius C.: Implement signing this message
-	_ = scbp.collectOutGoingOperations()
-
 	createdBlockBody := &block.Body{MiniBlocks: miniblocks}
 	createdBlockBody.MiniBlocks = append(createdBlockBody.MiniBlocks, postProcessMBs...)
+
+	err = scbp.createAndSetOutGoingMiniBlock(headerHandler, createdBlockBody)
+	if err != nil {
+		return nil, err
+	}
+
 	return scbp.applyBodyToHeader(headerHandler, createdBlockBody)
 }
 
-func (scbp *sovereignChainBlockProcessor) collectOutGoingOperations() []byte {
+func (scbp *sovereignChainBlockProcessor) createAndSetOutGoingMiniBlock(headerHandler data.HeaderHandler, createdBlockBody *block.Body) error {
 	logs := scbp.txCoordinator.GetAllCurrentLogs()
-	return scbp.outgoingOperationsFormatter.CreateOutgoingTxData(logs)
+	outGoingOperations := scbp.outgoingOperationsFormatter.CreateOutgoingTxsData(logs)
+	if len(outGoingOperations) == 0 {
+		return nil
+	}
+
+	outGoingMb, outGoingOperationsHash := scbp.createOutGoingMiniBlockData(outGoingOperations)
+	return scbp.setOutGoingMiniBlock(headerHandler, createdBlockBody, outGoingMb, outGoingOperationsHash)
+}
+
+func (scbp *sovereignChainBlockProcessor) createOutGoingMiniBlockData(outGoingOperations [][]byte) (*block.MiniBlock, []byte) {
+	outGoingOpHashes := make([][]byte, len(outGoingOperations))
+	aggregatedOutGoingOperations := make([]byte, 0)
+
+	for idx, outGoingOp := range outGoingOperations {
+		outGoingOpHash := scbp.hasher.Compute(string(outGoingOp))
+		aggregatedOutGoingOperations = append(aggregatedOutGoingOperations, outGoingOpHash...)
+
+		outGoingOpHashes[idx] = outGoingOpHash
+		scbp.outGoingOperationsPool.Add(outGoingOpHash, outGoingOp)
+	}
+
+	// TODO: We need to have a mocked transaction with this hash to be saved in storage and get rid of following warnings:
+	// 1. basePreProcess.createMarshalledData: tx not found hash = bf7e...
+	// 2. basePreProcess.saveTransactionToStorage  txHash = bf7e... dataUnit = TransactionUnit error = missing transaction
+	// Task for this: MX-14716
+	return &block.MiniBlock{
+		TxHashes:        outGoingOpHashes,
```

Comment:
You can create SCRs for this outGoingOperations in the OutGoingOps post processor.

Reply:
and those will be saved in the storage.